bundle/src/main/java/com/adobe/acs/commons/workflow/process/impl/CopyPropertiesProcess.java

```diff
@@ -0,0 +1,162 @@
+    private static final String SEPARATOR = "->";
+    private static final String ALTERNATE_SEPARATOR = "=>";
+    private static final String EVENT_DATA = "acs-aem-commons.workflow.copy-properties";
+
+    @Reference
+    private WorkflowPackageManager workflowPackageManager;
+
+    @Reference
+    private WorkflowHelper workflowHelper;
+
+    @Override
+    public final void execute(WorkItem workItem, WorkflowSession workflowSession, MetaDataMap metaDataMap) throws WorkflowException {
+        String wfPayload = null;
+
+        try (ResourceResolver resourceResolver = workflowHelper.getResourceResolver(workflowSession)) {
+            wfPayload = (String) workItem.getWorkflowData().getPayload();
+
+            final List<String> payloads = workflowPackageManager.getPaths(resourceResolver, wfPayload);
+
+            for (final String payload : payloads) {
+                copyProperties(metaDataMap, resourceResolver, payload);
+            }
+
+            if (resourceResolver.hasChanges()) {
+                setJcrSessionUserData(resourceResolver);
+            }
+        } catch (RepositoryException e) {
+            throw new WorkflowException(String.format("Could not find the payload for '%s'", wfPayload), e);
+        }
+    }
+
+    protected void setJcrSessionUserData(ResourceResolver resourceResolver) {
+        try {
+            resourceResolver.adaptTo(Session.class).getWorkspace().getObservationManager().setUserData(EVENT_DATA);
+        } catch (RepositoryException e) {
+            log.warn("Unable to set user-data to [ " + EVENT_DATA + " ]", e);
+        }
+    }
+
+    protected void copyProperties(MetaDataMap metaDataMap, ResourceResolver resourceResolver, String payload) {
+        final Resource resource = workflowHelper.getPageOrAssetResource(resourceResolver, payload);
+
+        if (resource == null) {
+            log.warn("Could not process payload [ {} ] as it could be resolved to a Page or Asset", payload);
+            return;
+        }
+
+        final Boolean skipEmptyValue = metaDataMap.get(PN_SKIP_EMPTY_SOURCE_PROPERTY, Boolean.TRUE);
+        final String[] propertyMaps = metaDataMap.get(PN_PROPERTY_MAP, new String[]{});
+
+        for (String propertyMap : propertyMaps) {
+            propertyMap = StringUtils.replace(propertyMap, ALTERNATE_SEPARATOR, SEPARATOR);
+            Map.Entry<String, String> entry = ParameterUtil.toMapEntry(propertyMap, SEPARATOR);
+
+            try {
+                PropertyResource source = new PropertyResource(StringUtils.trim(entry.getKey()), resource.getPath(), resourceResolver);
+                PropertyResource destination = new PropertyResource(StringUtils.trim(entry.getValue()), resource.getPath(), resourceResolver);
+
+                destination.setValue(source.getValue(), skipEmptyValue);
+            } catch (WorkflowException e) {
+                log.error("Could not copy properties", e);
```

Comment:
Can you add source and desination resource to that message? Otherwise it's hard to debug.